src/all/xiutaku/src/eu/kanade/tachiyomi/extension/all/xiutaku/Xiutaku.kt

```diff
@@ -0,0 +1,116 @@
+    override val name = "Xiutaku"
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient.newBuilder()
+        .rateLimitHost(baseUrl.toHttpUrl(), 10, 1, TimeUnit.SECONDS)
+        .setRandomUserAgent(UserAgentType.MOBILE)
+        .build()
+
+    override fun headersBuilder() = super.headersBuilder().add("Referer", "$baseUrl/")
+
+    // Latest
+    override fun latestUpdatesFromElement(element: Element) = SManga.create().apply {
+        thumbnail_url = element.selectFirst("img")?.attr("abs:src")
+        title = element.selectFirst(".item-content .item-link")!!.text()
+        setUrlWithoutDomain(element.selectFirst(".item-content .item-link")!!.attr("abs:href"))
+    }
+
+    override fun latestUpdatesNextPageSelector() = ".pagination-next:not([disabled])"
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/?start=${20 * (page - 1)}", headers)
+    }
+
+    override fun latestUpdatesSelector() = ".blog > div"
+
+    // Popular
+    override fun popularMangaFromElement(element: Element) = latestUpdatesFromElement(element)
+    override fun popularMangaNextPageSelector() = latestUpdatesNextPageSelector()
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/hot?start=${20 * (page - 1)}", headers)
+    }
+
+    override fun popularMangaSelector() = latestUpdatesSelector()
+
+    // Search
+    override fun searchMangaFromElement(element: Element) = latestUpdatesFromElement(element)
+    override fun searchMangaNextPageSelector() = latestUpdatesNextPageSelector()
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        return GET(
+            baseUrl.toHttpUrl().newBuilder().apply {
+                addQueryParameter("search", query)
+                addQueryParameter("page", (20 * (page - 1)).toString())
+            }.build(),
+            headers,
+        )
+    }
+
+    override fun searchMangaSelector() = latestUpdatesSelector()
+
+    // Details
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        title = document.selectFirst(".article-header")!!.text()
+        description = document.selectFirst(".article-info:not(:has(small))")?.text()
+        genre = document.selectFirst(".article-tags")
+            ?.select(".tags > .tag")?.joinToString { it.text().removePrefix("#") }
+    }
+
+    override fun chapterListSelector() = throw UnsupportedOperationException()
+    override fun chapterFromElement(element: Element) = throw UnsupportedOperationException()
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        val dateUploadStr = document.selectFirst(".article-info > small")?.text()
+        val dateUpload = DATE_FORMAT.tryParse(dateUploadStr)
+        val maxPage =
+            document.selectFirst(".pagination-list > span:last-child > a")?.text()?.toInt()
```

Comment:
I'm guessing you want to call back to the default if it can't parse the text as an integer.
```suggestion
            document.selectFirst(".pagination-list > span:last-child > a")?.text()?.toIntOrNull()
```

Reply:
updated